docs/spec.md

```diff
@@ -232,6 +236,41 @@ Informs the client about services that are no longer available. Only supported i
 }
 ```
 
+### Connection graph update
+
+Informs the client about updates to the connection graph. This is only sent to clients which have previously [subscribed to connection graph updates](#subscribe-connection-graph-updates). Only supported if the server previously declared that it has the `connectionGraph` [capability](#server-info).
+
+#### Fields
+
+- `op`: string `"connectionGraphUpdate"`
+- `publishedTopics`: array of
+  - `name`: topic name
+  - `publisherIds`: array of string, list of publisher IDs
+- `subscribedTopics`: array of
+  - `name`: topic name
+  - `subscriberIds`: array of string, list of subscriber IDs
+- `advertisedServices`: array of
+  - `name`: service name
+  - `providerIds`: array of string, list of provider IDs
+- `removedTopics`: array of string, names of topics that have been removed
+- `removedServices`: array of string, names of services that have been removed
```

Comment:
Need to clarify the semantics of these – do they replace all the data from the previous message? Do they only replace on an individual topic level? Or are they purely additive?